services/audit-service/src/main/java/datawave/microservice/audit/health/rabbit/RabbitHealthChecker.java

```diff
@@ -0,0 +1,705 @@
+            default:
+                throw new RuntimeException("Unable to create binding given properties: [" + bp + "]");
+        }
+        return new Binding(bp.getDestination(), destType, bp.getExchange(), bp.getRoutingKey(), bp.getArguments());
+    }
+    
+    /**
+     * Determines which poll interval should be used, based on the overall RabbitMQ health
+     * 
+     * @return the poll interval to use in millis
+     */
+    @Override
+    public long pollIntervalMillis() {
+        if (isHealthy())
+            return rabbitHealthProperties.getHealthyPollIntervalMillis();
+        else
+            return rabbitHealthProperties.getUnhealthyPollIntervalMillis();
+    }
+    
+    /**
+     * Performs a health check of RabbitMQ.
+     * <p>
+     * This method is synchronized with the isHealthy method to ensure that the health check is given time to complete before deciding whether to accept an
+     * audit request via the {@link AuditController}
+     * <p>
+     * RabbitMQ is considered to be healthy if the cluster, exchanges, queues, and bindings are configured as specified in the {@link RabbitHealthProperties}.
+     * If we are not able to determine the health of the cluster, exchanges, queues, and bindings (perhaps due to the management API being unavailable, or the
+     * cluster being down), we will consider RabbitMQ to be unhealthy.
+     */
+    @Override
+    public void runHealthCheck() {
+        initRabbitClient();
+        
+        synchronized (this) {
+            log.trace("RabbitMQ Health Check - Started");
+            
+            boolean wasHealthy = isHealthy();
+            
+            // check the cluster
+            clusterHealthCheck();
+            
+            // check the exchanges
+            exchangesHealthCheck();
+            
+            // check the queues
+            queuesHealthCheck();
+            
+            // check the bindings
+            bindingsHealthCheck();
+            
+            boolean isHealthy = isHealthy();
+            
+            if (wasHealthy) {
+                if (isHealthy) {
+                    log.info("RabbitMQ is still healthy.");
+                    lastSuccessfulHealthCheck = new Date();
+                } else {
+                    log.warn("RabbitMQ is not healthy.");
+                    currentOutageStats = new RabbitOutageStats(lastSuccessfulHealthCheck);
+                    outageStats.add(currentOutageStats);
+                    updateOutageStats(currentOutageStats);
+                }
+            } else {
+                if (isHealthy) {
+                    lastSuccessfulHealthCheck = new Date();
+                    if (currentOutageStats != null) {
+                        currentOutageStats.setStopDate(lastSuccessfulHealthCheck);
+                        currentOutageStats = null;
+                        log.info("RabbitMQ has recovered.");
+                    } else
+                        log.info("RabbitMQ is healthy");
```

Comment:
Nit: if you have braces on the if, then they should be on the else as well.

Reply:
Updated!